src/test/java/net/snowflake/client/config/SFPermissionsTest.java

```
@@ -0,0 +1,98 @@
+            put("rwxrwx---", false);
+            put("rwxrw----", false);
+            put("rwxr-x---", true);
+            put("rwxr-----", true);
+            put("rwx-wx---", false);
+            put("rwx-w----", false);
+            put("rwx--x---", true);
+            put("rwx---rwx", false);
+            put("rwx---rw-", false);
+            put("rwx---r-x", true);
+            put("rwx---r--", true);
+            put("rwx----wx", false);
+            put("rwx----w-", false);
+            put("rwx-----x", true);
+          }
+        };
+    return testConfigFilePermissions.entrySet();
+  }
+
+  Path configFilePath = Paths.get("config.json");
+  String configJson = "{\"common\":{\"log_level\":\"debug\",\"log_path\":\"logs\"}}";
+  String permission;
+  Boolean isSucceed;
+
+  public SFPermissionsTest(Map.Entry<String, Boolean> permission) {
+    this.permission = permission.getKey();
+    this.isSucceed = permission.getValue();
+  }
+
+  @Before
+  public void createConfigFile() {
+    try {
+      Files.write(configFilePath, configJson.getBytes());
+    } catch (IOException e) {
+      fail("testLogDirectoryPermissions failed setup");
+    }
+  }
+
+  @After
+  public void cleanupConfigFile() {
+    try {
+      Files.deleteIfExists(configFilePath);
+    } catch (IOException e) {
+      fail("testLogDirectoryPermissions failed cleanup");
+    }
+  }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnWin.class)
+  public void testLogDirectoryPermissions() {
+    // Don't run on Windows
+    try {
+      Files.setPosixFilePermissions(configFilePath, PosixFilePermissions.fromString(permission));
+      SFClientConfigParser.loadSFClientConfig(configFilePath.toString());
+      if (!isSucceed) {
+        fail("testLogDirectoryPermissions failed. Expected exception.");
+      }
+    } catch (IOException e) {
+      if (isSucceed) {
+        fail("testLogDirectoryPermissions failed. Expected pass.");
+      }
+    } catch (Exception e) {
```

we can remove this catch and let the method fail